resources/zib/zib-MedicationAgreement.xml

```diff
@@ -0,0 +1,269 @@
+    <telecom>
+      <system value="email" />
+      <value value="info@nictiz.nl" />
+      <use value="work" />
+    </telecom>
+  </contact>
+  <description value="A medication agreement is a prescriber’s proposal for a patient to use medication. An agreement to discontinue the use of medication is also a medication agreement." />
+  <purpose value="This MedicationRequest resource represents the Dutch [zib ('Zorginformatiebouwsteen', i.e. Health and Care Information Model) MedicationAgreement-v1.2(2020EN)](https://zibs.nl/wiki/MedicationAgreement-v1.2(2020EN))." />
+  <copyright value="Copyright and related rights waived via CC0, https://creativecommons.org/publicdomain/zero/1.0/. This does not apply to information from third parties, for example a medical terminology system. The implementer alone is responsible for identifying and obtaining any necessary licenses or authorizations to utilize third party IP in connection with the specification or otherwise." />
+  <fhirVersion value="4.0.1" />
+  <mapping>
+    <identity value="zib-medicationagreement-v1.2-2020EN" />
+    <uri value="https://zibs.nl/wiki/MedicationAgreement-v1.2(2020EN)" />
+    <name value="zib MedicationAgreement-v1.2(2020EN)" />
+  </mapping>
+  <kind value="resource" />
+  <abstract value="true" />
+  <type value="MedicationRequest" />
+  <baseDefinition value="http://hl7.org/fhir/StructureDefinition/MedicationRequest" />
+  <derivation value="constraint" />
+  <differential>
+    <element id="MedicationRequest">
+      <path value="MedicationRequest" />
+      <short value="MedicationAgreement" />
+      <alias value="Medicatieafspraak" />
+      <mapping>
+        <identity value="zib-medicationagreement-v1.2-2020EN" />
+        <map value="NL-CM:9.6.9580" />
+        <comment value="MedicationAgreement" />
+      </mapping>
+    </element>
+    <element id="MedicationRequest.extension">
+      <path value="MedicationRequest.extension" />
+      <slicing>
+        <discriminator>
+          <type value="value" />
+          <path value="url" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+    </element>
+    <element id="MedicationRequest.extension:additionalInformation">
+      <path value="MedicationRequest.extension" />
+      <sliceName value="additionalInformation" />
+      <type>
+        <code value="Extension" />
+        <profile value="http://nictiz.nl/fhir/StructureDefinition/ext-MedicationAgreement.AdditionalInformation" />
+      </type>
+    </element>
+    <element id="MedicationRequest.extension:instructionsForUseDescription">
+      <path value="MedicationRequest.extension" />
+      <sliceName value="instructionsForUseDescription" />
+      <max value="1" />
+      <type>
+        <code value="Extension" />
+        <profile value="http://nictiz.nl/fhir/StructureDefinition/ext-InstructionsForUse.Description" />
+      </type>
+    </element>
+    <element id="MedicationRequest.extension:periodOfUse">

```

Comment:
The zib describes the PeriodOfUse concept as:

> [...] This is the time at which the instructions for use in this agreement start. [...]

This seems to suggest that `Timing.event` and/or `Timing.repeat.boundsPeriod` might be appropriate elements to convey this information (and that this definition overlaps with DosageDuration from InstructionsForUse). Would this be an option?

Reply:
`.dosageInstruction` should be restricted to 0..1, there can be max. 1 InstructionsForUse. The concept description makes it clear that this information is directly coupled with this InstructionsForUse, so IMO it makes sense to communicate it using the elements that are defined for this purpose.

Reply:
`.dosageInstruction` cannot be set to 0..1. Otherwise DosingInstructions (0..*) within InstructionsForUse cannot be mapped.

Reply:
I tried to explain this on the root of the InstructionForUse profile.

Reply:
I'm afraid you're right ...